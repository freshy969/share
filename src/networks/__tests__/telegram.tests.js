import Telegram from '../telegram';

describe('Share > Telegram', () => {
  test('url - all parameters', () => {
    const telegram = Telegram.create({});
    Object.defineProperty(telegram, 'entityLink', {value: jest.fn(), writable: true});
    telegram.entityLink.mockReturnValueOnce('https://demo.frontity.com/the-beauties-of-gullfoss');

    expect(
      telegram.url({
        type: 'post',
        id: 60,
        text: 'The Beauties of Gullfoss',
      }),
    ).toMatchSnapshot();
  });

  test('url - required parameters', () => {
    const telegram = Telegram.create({});
    Object.defineProperty(telegram, 'entityLink', {value: jest.fn(), writable: true});
    telegram.entityLink.mockReturnValueOnce('https://demo.frontity.com/the-beauties-of-gullfoss');

    expect(
      telegram.url({
        type: 'post',
        id: 60,
      }),
    ).toMatchSnapshot();
  });
});
