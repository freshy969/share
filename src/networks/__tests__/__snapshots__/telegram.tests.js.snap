// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share > Telegram url - all parameters 1`] = `"https://telegram.me/share?url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss&text=The%20Beauties%20of%20Gullfoss"`;

exports[`Share > Telegram url - required parameters 1`] = `"https://telegram.me/share?url=https%3A%2F%2Fdemo.frontity.com%2Fthe-beauties-of-gullfoss"`;
